Bulk Plugins Management: Adds list view

cpapazoglou opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes https://github.com/Automattic/dotcom-forge/issues/9854

## Proposed Changes

* Used `list` type for the plugins list dataview when opening the sites list pane or when viewport is smaller than `960px`.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* The changes aim to make the navigation friendlier, especially in mobile viewports 

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->
 
* Go to `/plugins/manage/sites`
* See how the plugin list changes to list view and back to table view when you navigate to a specific plugin or when the viewport gets < 960px

|Before | After|
|-------|------|
|![CleanShot 2025-01-13 at 21 11 11](https://github.com/user-attachments/assets/aff84c2d-1ce5-4e59-8cde-ebb2975c953f)|![CleanShot 2025-01-13 at 21 11 25](https://github.com/user-attachments/assets/7d7d23df-b2b2-48ed-92e5-a1dbc51f75d3)|
|![CleanShot 2025-01-13 at 21 26 20](https://github.com/user-attachments/assets/a68e571d-3771-4333-8c2b-2dbcae366b28)|![CleanShot 2025-01-13 at 21 26 31](https://github.com/user-attachments/assets/f7632e52-dea1-4a93-a251-dcb7297537ff)|


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

cpapazoglou committed:
Adds mobile view

Commit:
Plugins Dataviews: switches from table to list automatically.

Commit:
Plugins Dataview: automatically switches to list

epeicher reviewed (approved):
Thanks for applying the changes, I think this greatly improves the small viewports views. I have reviewed the changes and they LGTM. I have a couple of non-blocking comments

1. When clicking on an item of the list, then closing and then resizing again. I have observed a weird behaviour. This is so specific that I consider this non-blockinng. Please see a video 

https://github.com/user-attachments/assets/7fa14431-f816-476b-9ebf-b907f6c18f3a

2. Would it be possible to open the plugins when selecting anywhere in the cell? It was a bit confusing to me to see the `pointer` cursor but not been able to click. I see that this already happens in the table view so it is not related to this specific PR, but commenting here in case is something that can be addressed as part of this or another PR.
![CleanShot 2025-01-15 at 12 09 34@2x](https://github.com/user-attachments/assets/2353b65e-8cce-429d-a18a-549532b1d22d)

Commit:
Fixes visilble fields.

cpapazoglou commented:
> Thanks for applying the changes, I think this greatly improves the small viewports views. I have reviewed the changes and they LGTM. I have a couple of non-blocking comments
> 
> 1. When clicking on an item of the list, then closing and then resizing again. I have observed a weird behaviour. This is so specific that I consider this non-blockinng. Please see a video
> 
>  CleanShot.2025-01-15.at.11.54.58.mp4 
> 2. Would it be possible to open the plugins when selecting anywhere in the cell? It was a bit confusing to me to see the `pointer` cursor but not been able to click. I see that this already happens in the table view so it is not related to this specific PR, but commenting here in case is something that can be addressed as part of this or another PR.
>    ![CleanShot 2025-01-15 at 12 09 34@2x](https://private-user-images.githubusercontent.com/3519124/403391863-2353b65e-8cce-429d-a18a-549532b1d22d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwNDExNzIsIm5iZiI6MTczNzA0MDg3MiwicGF0aCI6Ii8zNTE5MTI0LzQwMzM5MTg2My0yMzUzYjY1ZS04Y2NlLTQyOWQtYTE4YS01NDk1MzJiMWQyMmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDExNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMTZUMTUyMTEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDBmN2VjMDliMTJkNjNkYThmYTllM2RkOTJmYjZhOGEwMGRhMjg3Yjg5YTM0ODQwZDQ1YjQ2YjVmMDA3ZmVlMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.2KVL7gbSoN1liHZxRdhmLKwF8Q9RaOW45hXCpS8H9WM)

I have fixed the first issue. Regarding the 2nd issue with the pointer, it's a bit hard making the whole `li` clickable as it currently contains the "Update" link and the elipsis menu in mobile views. I have removed the `cursor: pointer` though which made it feel like something is broken.

Commit:
Removes cursor pointer from list view.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17196237

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @cpapazoglou for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17196237).
